test/unit/backend_api_logic/metric_extension_test.rb

```diff
@@ -11,17 +11,12 @@ class MetricExtensionTest < ActiveSupport::TestCase
   attr_reader :backend_api, :metric
 
   test 'extends metric system_name with backend api id' do
-    extended_system_name = metric.extended_system_name
+    extended_system_name = metric.send(:extended_system_name)
```

Comment:
I do not think that the response of `extended_system_name` matters enough to be tested on its own anymore considering that now it is a protected method. What truly matters now is what it is saved in `attributes['system_name']` and what `system_name` returns imo 🤔 

So I'd go for testing like this:
```ruby
test 'extends metric system_name with backend api id' do
  assert metric.save
  assert_equal "whatever.#{backend_api.id}", metric.reload.attributes['system_name']
  assert_equal "whatever.#{backend_api.id}", metric.system_name
end
```